src/caching/Arweave/ArweaveClient.ts

```diff
@@ -0,0 +1,110 @@
+// import { Struct } from "superstruct";
+// import { CachingMechanismInterface } from "../../interfaces";
+import Arweave from "arweave";
+import { JWKInterface } from "arweave/node/lib/wallet";
+import { ethers } from "ethers";
+import winston from "winston";
+import { jsonReplacerWithBigNumbers, parseWinston } from "../../utils";
+import { Struct, is } from "superstruct";
+
+export class ArweaveClient {
+  private client: Arweave;
+
+  public constructor(
+    private arweaveJWT: JWKInterface,
+    private logger: winston.Logger,
+    gatewayURL = "arweave.net",
+    protocol = "https",
+    port = 443
+  ) {
+    this.client = new Arweave({
+      host: gatewayURL,
+      port,
+      protocol,
+      timeout: 20000,
+      logging: false,
+    });
+    this.logger.info("Arweave client initialized");
+  }
+
+  /**
+   * Stores an arbitrary record in the Arweave network. The record is stored as a JSON string and uses
+   * JSON.stringify to convert the record to a string. The record has all of its big numbers converted
+   * to strings for convenience.
+   * @param value The value to store
+   * @returns The transaction ID of the stored value
+   * @
+   */
+  async set(value: Record<string, unknown>): Promise<string | undefined> {
+    const transaction = await this.client.createTransaction(
+      { data: JSON.stringify(value, jsonReplacerWithBigNumbers) },
+      this.arweaveJWT
+    );
+    // Add tags to the transaction
+    transaction.addTag("Content-Type", "application/json");
+    // Sign the transaction
+    await this.client.transactions.sign(transaction, this.arweaveJWT);
+    // Send the transaction
+    const result = await this.client.transactions.post(transaction);
```

Comment:
how do fees work?

Reply:
Yes we should send an error to the caller if there aren't funds to pay for the txn